File: src/dotnet/Kurdi.ECommerce.Inventory.Core/Entities/ProductAggregate/Product.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Kurdi.ECommerce.Inventory.Core.Entities.CategoryAggregate;

namespace Kurdi.ECommerce.Inventory.Core.Entities.ProductAggregate
{
    [Table(name: "products")]
    public class Product
    {
        [Key]
        [Column(name: "sku")]
        public string SKU { get; set; }
        [Column(name: "supplier_identity")]
        public int SupplierIdentity { get; set; }
        public ProductPrices ProductPrices { get; set; }
        public ProductQuantity ProductQuantity { get; set; }
        public List<ProductDetails> ProductDetails { get; set; } = new List<ProductDetails>();
        [Column(name: "category")]
        public string CategoryName { get; set; }
        [ForeignKey(name: "CategoryName")]
        public Category Category { get; set; }
        [Column(name: "activation")]
        public bool Activation { get; set; }
        public TimeStamps TimeStamps { get; set; }

    }
}

File: inventory/src/main/java/com/kurdi/ecommerce/inventoryservice/domain/entities/products/Product.java
package com.kurdi.ecommerce.inventoryservice.domain.entities.products;

import com.kurdi.ecommerce.inventoryservice.domain.entities.categories.Category;
import com.kurdi.ecommerce.inventoryservice.domain.entities.products.details.ProductDetails;
import lombok.*;
import org.hibernate.Hibernate;
import org.hibernate.id.GUIDGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.util.*;


@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "stock_items")
public class Product implements Serializable {
    @Id
    @Builder.Default
    String sku = new GUIDGenerator().toString();
    Integer SupplierIdentity;
    @Embedded
    ProductPrices stockItemPrices;
    @Embedded
    ProductQuantity stockItemQuantity;
    @ManyToOne(fetch = FetchType.EAGER)
    @Builder.Default
    Category category = new Category();
    @OneToMany(mappedBy = "stockItemDetailsId.stockItem", fetch = FetchType.EAGER, cascade = CascadeType.PERSIST)
    @Builder.Default
    List<ProductDetails> stockItemDetails = new ArrayList<>();


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Product stockItem = (Product) o;
        return sku != null && Objects.equals(sku, stockItem.sku);
    }
    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
